be/src/exec/vectorized/partition/chunks_partitioner.h

```diff
@@ -39,7 +40,31 @@ class ChunksPartitioner {
     // method signature is: bool consumer(int32_t partition_idx, const ChunkPtr& chunk)
     // The return value of the consumer denote whether to continue or not
     template <typename Consumer>
-    Status accept(Consumer&& consumer);
+    Status accept(Consumer&& consumer) {
+        // First, fetch chunks from hash map
+        if (false) {
+        }
+#define HASH_MAP_METHOD(NAME)                                                                                    \
+    else if (_hash_map_variant.type == PartitionHashMapVariant::Type::NAME) {                                    \
+        TRY_CATCH_BAD_ALLOC(fetch_chunks_from_hash_map<typename decltype(_hash_map_variant.NAME)::element_type>( \
+                *_hash_map_variant.NAME, consumer));                                                             \
+    }
+        APPLY_FOR_PARTITION_VARIANT_ALL(HASH_MAP_METHOD)
+#undef HASH_MAP_METHOD
+
+        // Second, fetch chunks from null_key_value if any
+        if (false) {
+        }
+#define HASH_MAP_METHOD(NAME)                                                                                          \
+    else if (_hash_map_variant.type == PartitionHashMapVariant::Type::NAME) {                                          \
+        TRY_CATCH_BAD_ALLOC(fetch_chunks_from_null_key_value<typename decltype(_hash_map_variant.NAME)::element_type>( \
```

Comment:
There is a cost to move and compare items in chunk. If it is NullFirst, when there are many Null values, placing it in front may avoid the movement of some items of the Column int the TopN Stage.

Reply:
Can be optimized later